### DISCUSSION TITLE
GitHub Actions | Listing workflows cancelled by concurrency settings as failed could be misleading

### DISCUSSION THREAD

**Discussion by GabrielCousin** [community/community#8336] [Actions]
Hi!

We noticed `concurrency` settings were introduced lately in GitHub actions and we gave them a try. However, we chose not to use them at the moment for the following reason:

As shown in the below screenshot, the penultimate commit is listed as failed (because its workflow was canceled by the last one)
The cancel feature is working as expected but the team considered it was misleading to see a red cross on it (if you click for more details, you can finally understand it was canceled though).

<img width="772" alt="Screenshot 2021-12-02 at 10 32 59" src="https://user-images.githubusercontent.com/7649529/144395393-322a237f-9aa8-4919-8f1e-52b882ca9490.png">

<img width="389" alt="Screenshot 2021-12-02 at 10 36 13" src="https://user-images.githubusercontent.com/7649529/144396099-37450a8b-ee6b-46dc-b870-5c294cbee1af.png">

We don't know if an option can be introduced not to consider these as failed or maybe introducing another icon.

Thanks a bunch!

**Comment by lars-reimann** [comment]
I'd like this as well. Something like an optional property `treat-cancellation-as` with values `success` and `failure` (default) under `concurrency` and `jobs.<job_id>.concurrency` would be great.

Generally, there is some mismatch between the symbols in the commit history for all checks taken together and the symbols for individual checks:

In the history there are symbols for
* Success
* Failure
* Pending

For a single check there are symbols for
* Success
* Failure
* Pending
* Cancelled
* Skipped

In the history a skip has the same effect as a success, while cancellation has the same effect as a failure. The true result is not immediately obvious without clicking on the commit. There should maybe be additional symbols shown in the commit history to denote
* some jobs were skipped, the rest was successful
* some jobs were cancelled, the rest was sucessful or skipped

**Comment by ethomson** [comment]
Thanks @GabrielCousin - I can see how this could be confusing.  I'll take a look at how we can improve this.

**Comment by rodo-r2r** [comment]
We're seeing the same issue after adding `concurrency` to our CI.
It would be great if we could treat this as not being an error.

**Comment by rocklan** [comment]
I'm seeing this as well, it would be great if it could be improved.

**Comment by clong-** [comment]
We're seeing this too. I think that if workflows are designed around this feature, then the designer is probably acknowledging that cancellation is an acceptable state, hence should not necessarily be treated as failure. It would be great to see a configurable option to dictate whether cancellation yields a failure state or is ignored.

**Comment by idejanovic** [comment]
One option would be to have another option than the `cancel-in-progress` like:

```
concurrency:
  group: "..."
  skip-in-progress: true
```

**Comment by max-sixty** [comment]
This also causes badges in the repo to list the build as failing, e.g. in [PRQL](https://github.com/prql/prql), after a PR was merged while a previous workflow was running on main:

<img width="268" alt="image" src="https://user-images.githubusercontent.com/5635139/190015617-54dde872-db69-4bc5-8274-3267999c444c.png">

Is there any workaround for this specific instance of the issue? e.g. has anyone tried making an `if` expression which generates something unique on the main branch?

**Comment by sanity** [comment]
Experiencing this same problem, anyone have a workaround?

**Comment by goodsign** [comment]
Any news on that? Because of this, most of the builds in our CI are now red and the team is getting used to the fact that red CI is normal and usual.

**Comment by leahy268** [comment]
As above we have the same problem.  Multiple Teams deploying and we have PreProd and Prod environments that require approval,  We then have these set to "cancel-in-progress: true".  This is deliberate as without that pending deployments will just eventually timeout.  We only want the latest deployment to be going into Prod and PreProd so the cancel-in-progress true achieves this beautifullly.  

However all of these deployments then appear as failed when everything was successful except that we chose not to let a deployment go through to prod.

**Comment by torokati44** [comment]
+1

**Comment by mpushki** [comment]
+1

**Comment by gustavovalverde** [comment]
Hopefully someone at GitHub is looking at this as the Developer Experience is annoying with all thoses ❌'s, making the team used to see those as an _"expected outcome"_, and makes other things very confusing, like notifications on our `main` branch when something fails —as those are not actual failures, but gets reported as one—.

**Comment by Mine02C4** [comment]
+1

**Comment by mpushki** [comment]
I use this name for group
```
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
  cancel-in-progress: true
```

**Comment by davinchia** [comment]
Still no solution for this? We are a big open source repo and are also running into this problem.

**Comment by psyinf** [comment]
I'm frustrated by this, too. There is no obvious way how to change how the check-summary is evaluated (e.g. not getting "Some checks were not successful" and the red cross in the summary view. 
This needs to be either an explicit per-repo configuration option (treat_canceled_as_success) or an option within the jobs
For me the issue arises due to having a two jobs in the pipeline for pushes on feature-branches and updates to PRs which don't need to be performed both in case there is a PR.

**Comment by dep** [comment]
Need a solution for this too.

**Comment by bmson** [comment]
Second on this issue. The visual indication of failed tests is affected by intentionally canceled workflows showing up as failed.

Canceled workflows should have a different indicator

**Comment by gabe-mercury** [comment]
+1

**Comment by Gabriella439** [comment]
Our company open sourced a GitHub action you can use to work around this problem:

https://github.com/MercuryTechnologies/delete-cancelled-runs

… but it would be really nice if there were a GitHub-native solution to this problem.  This GitHub action of ours is more of a hack than a proper solution.

**Comment by thisisnacho** [comment]
+1

**Comment by zwang1-godaddy** [comment]
+1

**Comment by marcaddeo** [comment]
I cancel workflows with the concurrency rules, and just using login in automated workflows. Having everything show as a red X when they're intentionally being cancelled doesn't make much sense.

**Comment by jaredscheib** [comment]
We also see this as a Failure in Slack via the GitHub bot, which is misleading and confusing every time, contributing to alert fatigue. GitHub and the bot should report this as a Cancellation, not a Failure.

Any update from GitHub? This issue's been open for 5+ years now.
